private/bufpkg/bufplugin/bufpluginconfig/bufpluginconfig.go

```diff
@@ -293,7 +346,51 @@ type ExternalNPMRegistryConfig struct {
 
 // ExternalMavenRegistryConfig is the external registry configuration for a Maven plugin.
 type ExternalMavenRegistryConfig struct {
+	Compiler           ExternalMavenCompilerConfig `json:"compiler" yaml:"compiler"`
+	Deps               []string                    `json:"deps,omitempty" yaml:"deps,omitempty"`
+	AdditionalRuntimes []ExternalMavenRuntime      `json:"additional_runtimes,omitempty" yaml:"additional_runtimes,omitempty"`
+}
+
+// ExternalMavenCompilerConfig configures compiler settings for Maven remote packages.
+type ExternalMavenCompilerConfig struct {
+	Java   ExternalMavenCompilerJavaConfig   `json:"java" yaml:"java"`
+	Kotlin ExternalMavenCompilerKotlinConfig `json:"kotlin" yaml:"kotlin"`
+}
+
+// ExternalMavenCompilerJavaConfig configures the Java compiler settings for remote packages.
+type ExternalMavenCompilerJavaConfig struct {
+	// Encoding specifies the encoding of the source files (default: UTF-8).
+	Encoding string `json:"encoding" yaml:"encoding"`
+	// Release specifies the target Java release (default: 8).
+	Release int `json:"release" yaml:"release"`
+	// Source specifies the source bytecode level (default: 8).
+	Source int `json:"source" yaml:"source"`
+	// Target specifies the target bytecode level (default: 8).
+	Target int `json:"target" yaml:"target"`
+}
+
+// ExternalMavenCompilerKotlinConfig configures the Kotlin compiler settings for remote packages.
+type ExternalMavenCompilerKotlinConfig struct {
+	// APIVersion specifies the Kotlin API version to target.
+	APIVersion string `json:"api_version" yaml:"api_version"`
+	// JVMTarget specifies the target version of the JVM bytecode (default: 1.8)
+	JVMTarget string `json:"jvm_target" yaml:"jvm_target"`
+	// LanguageVersion is used to provide source compatibility with the specified Kotlin version.
+	LanguageVersion string `json:"language_version" yaml:"language_version"`
+	// Version of the Kotlin compiler to use (required for Kotlin plugins).
+	Version string `json:"version" yaml:"version"`
+}
+
+// ExternalMavenRuntime allows configuring additional runtimes for remote packages.
+// These can specify different dependencies and compiler options than the default runtime.
+// This is used to support a single plugin supporting both full and lite Protobuf runtimes.
+type ExternalMavenRuntime struct {
```

Comment:
should we make this `ExternalMavenRuntimeConfig` to be inline with the other structs here?

Reply:
Fixed.